### PR TITLE
Feat: Controller specs

### PR DESCRIPTION
**Pull request by rubydevi**
### Description
In this PR, I have set up tests on the controllers.

### Changes made

- Added RSpec as a gem to my project.
- Created a Request spec file for all controllers.
- Made sure that for each action I check:
   - If response status was correct.
   - If a correct template was rendered.
   - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (topeogunleye) [inline spec/requests/posts_spec.rb:3]: Hello @rubydevi,

[OPTIONAL]
Great job on your RSpec tests! 🎉 I can see you've been diligent in ensuring that each action’s response status, rendered template, and response body meet the project requirements. Your attention to detail and commitment to quality is commendable.

### **Optional Improvement:**
Consider organizing your tests a bit more semantically by grouping them based on the actions of your controllers. For instance, you could structure your tests like so:

```ruby
describe 'GET /index' do
  # Your index action tests go here
  ...
end

describe 'GET /show' do
  # Your show action tests go here
  ...
end
```

This approach makes your test file more readable and allows other developers (or even yourself, in the future) to quickly understand the structure of your tests and find what they're looking for.

Here’s a possible refactoring of your tests using this structure:

```ruby
RSpec.describe 'Posts', type: :request do
  
  describe 'GET /users/123/posts' do
    context 'index action' do
      ...
    end
  end
  
  describe 'GET /users/123/posts/1' do
    context 'show action' do
      ...
    end
  end
end
```

Again, this is optional and just a suggestion for improving the readability of your tests. Your current structure is already well-organized and meets the project requirements effectively.

Keep up the excellent work, and happy coding! 🌟

**Review by topeogunleye — APPROVED** [review]
### **STATUS: APPROVED  ✔️** 
------
Hi @rubydevi,  

Your project is complete! Great job for nailing it right on the first attempt 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟

wiserhood-cheers

## To highlight :champagne: 
- Incorporation of RSpec Gem: :heavy_check_mark: 
- Comprehensive Request Spec Files :heavy_check_mark: 
- Thorough Action Checks :heavy_check_mark:
- Code Quality:  is clear, concise, and well-organized :heavy_check_mark:

## [Optional] suggestions:
 * [ ]  Please check my inline comments.

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear
Please, remember to tag @topeogunleye in your question so I can receive the notification

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (rubydevi) [reply spec/requests/posts_spec.rb:3]: Thanks a lot @topeogunleye for your constructive feedback, I appreciate it and I do agree with you on making it more readable. I will definitely make the recommended changes and merge it 😄 Thanks! Have a lovely day.

Comment (topeogunleye) [reply spec/requests/posts_spec.rb:3]: @rubydevi you're welcome

**Pull request merged**